syft/generic/abstract/message_handler.py

```diff
@@ -0,0 +1,20 @@
+from abc import ABC
+from abc import abstractmethod
+
+from syft.generic.object_storage import ObjectStore
+
+
+class AbstractMessageHandler(ABC):
+    def __init__(self, object_store: ObjectStore):
+        self.object_store = object_store
+        self.routing_table = self.init_routing_table()
+
+    @abstractmethod
+    def init_routing_table(self):
+        return {}
+
+    def supports(self, msg):
+        return type(msg) in self.routing_table.keys()
+
+    def handle(self, msg):
+        return self.routing_table[type(msg)](msg)
```

Comment:
Why not raising an error here if type(msg) isn't in the routing table and just catch it in workers to see what to do with it?

Reply:
I considered that, but then I remembered _The_ ~~10~~ 6 _Commandments Of PySyft Development_:

1. Thou shalt make everything a tensor type.
2. Thou shalt not implement messaging systems that circumvent `BaseWorker`.
3. Thou shalt not send data without using either `.send()`, `.get()`, or `.move()`.
4. Thou shalt not use `exec()`.
5. **Thou shalt not implement logic using try/except unless the logic itself does not call anywhere else in PySyft's codebase (purely external libraries.)**
6. Thou shalt never expose context managers to the user.

For the record, I don't make these rules. These were all @iamtrask. 😂